### ISSUE TITLE
bug: DateScalar repr says it's a Timestamp

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#7299]
### What happened?

`ibis.date("2000-01-01")` presents itself to `Timestamp('2000-01-01 00:00:00')`. This really confused me for a while.

This is different from if you have a column, which advertises itself as a date:
```
ibis.memtable({"date": ["2020-01-01", "2020-01-02", "2020-01-03"]}).date.cast("date")
┏━━━━━━━━━━━━━━━━━━┓
┃ Cast(date, date) ┃
┡━━━━━━━━━━━━━━━━━━┩
│ date             │
├──────────────────┤
│ 2020-01-01       │
│ 2020-01-02       │
│ 2020-01-03       │
└──────────────────┘
```
I would expect the DateScalar to look like the column, perhaps `Date('2000-01-01')`

### What version of ibis are you using?

7.0.0

### What backend(s) are you using, if any?

duckdb

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
Ah, we are normalizing to `date`, but execution by the default backend produces a pandas Timestamp. We can look into make this return a `date`.

**Comment by jcrist** [comment]
I don't think this is specific to the duckdb backend - the `PandasType` format will handle date types by converting them to numpy datetime64 types: https://github.com/ibis-project/ibis/blob/c1463815161793654bd7d40d54564a0b1940af5c/ibis/formats/numpy.py#L70-L72. We could instead treat these as `object` columns of `datetime.date` instances instead, but this would change not only the scalar format but also the columnar format. :shrug: as to which is better. As one datapoint in favor of this change, Pyarrow's `Table.to_pandas` will return `date` types as object columns by default, you have to pass `date_as_object=False` to get a `datetime64` column.

**Comment by cpcloud** [comment]
+1 to moving to object date types, while less efficient they're also less confusing IMO

**Comment by cpcloud** [comment]
This uncovered some inconsistency in our date handling, making it a bit of a 🐰 🕳️

**Comment by cpcloud** [comment]
I think to fix this we're going to need to change the execution output type of date columns and scalars to be object (containing `datetime.date` objects when using `to_pandas()`) which is a breaking change.

I'm taking this issue off the 7.2 milestone.

**Comment by cpcloud** [comment]
Fixed in #7734, will be released in 8.0.0!
